quantlib/pricingengines/_bondfunctions.pxd

```diff
@@ -0,0 +1,64 @@
+include '../types.pxi'
+
+from quantlib.instruments._bonds cimport Bond as QLBond
+from quantlib.handle cimport shared_ptr
+from quantlib.time._period cimport Frequency
+from quantlib.termstructures._yield_term_structure cimport YieldTermStructure
+from quantlib.time._daycounter cimport DayCounter
+from quantlib.interest_rate cimport InterestRate
+
+cimport quantlib.time._date as _dt
+
+cdef extern from 'ql/cashflows/duration.hpp' namespace 'QuantLib':
+    ctypedef enum Type "QuantLib::Duration::Type":
+        Simple    "QuantLib::Duration::Simple"
+        Macaulay  "QuantLib::Duration::Macaulay"
+        Modified  "QuantLib::Duration::Modified"
+
+cdef extern from 'ql/cashflows/duration.hpp' namespace 'QuantLib':
+    cdef cppclass Duration:
+        Type type
+
+cdef extern from 'ql/pricingengines/bond/bondfunctions.hpp' namespace 'QuantLib':
+    cdef Rate _bf_yield "BondFunctions::yield" (QLBond, Real, DayCounter, int, Frequency, _dt.Date, Real, Size, Rate)
+
+cdef extern from 'ql/pricingengines/bond/bondfunctions.hpp' namespace 'QuantLib':
+
+    cdef cppclass BondFunctions:
```

Comment:
This works but is not correct. You should have all the functions defined as the `_bf_yield` function. That will also simplify the code on the bonfunctions.pyx file where you can have simple python functions and no need for a Cython Extension class.